docs/search/docsearch-scraper.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash -x
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: MIT
+
+build_docs=false
+
+while getopts a:bc:d:i:p:r:u: flag
+do
+    case "${flag}" in
+        a) api=${OPTARG};;
+        b) build_docs=true;;
+        c) config=${OPTARG};;
+        d) docs=${OPTARG};;
+        i) index=${OPTARG};;
+        p) port=${OPTARG};;
+        r) remote=${OPTARG};;
+        u) url=${OPTARG};;
+    esac
+done
+
+# Get the directory of the current script
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Set defaults
+if [ -z "$api" ]; then
+  api=xyz
+fi
+if [ -z "$config" ]; then
+  config="$script_dir/scraper-config.json"
+fi
+if [ -z "$docs" ]; then
+  docs="$script_dir/../../target/slintdocs/html"
+fi
+if [ -z "$index" ]; then
+  index=local
+fi
+if [ -z "$port" ]; then
+  port=8108
+fi
+if [ -n "$remote" ]; then
+  host=$remote
+  server=$remote
+  protocol=https
+  port=443
+  hostport=$remote
+else
+  host=localhost
+  server=host.docker.internal
+  protocol=http
+  port=$port
+  hostport=$host:$port
+fi
+if [ -z "$url" ]; then
+  url=http://localhost:8000
+fi
+
+# echo "Script Directory: $script_dir";
+# echo "API Key: $api";
+# echo "Config: $config";
+# echo "Docs: $docs";
+# echo "Host: $host";
+# echo "Index: $index";
+# echo "Port: $port";
+# echo "Protocol: $protocol";
+# echo "Server: $server";
+# echo "Url: $url";
+# echo "HostPort: $hostport";
+
```

Comment:
Preparation to make this work on macOS: (brew install sed to install gnu sed, alias gsed)

```suggestion

if [[ $OSTYPE == 'darwin'* ]]; then
  sed=gsed
else
  sed=sed
fi
```